### PR TITLE
Add `patch_radius` parameter to Patch2Self denoise workflow

### PR DESCRIPTION
**Pull request by pcamach2**
Addresses #2791

### CODE REVIEW

File: dipy/workflows/denoise.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def run(self, input_files, bval_files, model='ols',
             Regularization parameter only for ridge regression model.
         verbose : bool, optional
             Show progress of Patch2Self and time taken.
+        patch_radius : int or 1 D array, optional
```
Comment (skoudoro) [inline dipy/workflows/denoise.py:58]: `variable int, optional` instead 

you can look at this example: https://github.com/dipy/dipy/blob/976217e547c4da6bfb210642adbb071c2ee40dd9/dipy/workflows/denoise.py#L277

**Review by skoudoro — CHANGES_REQUESTED** [review]
Nice! Thank you for this @pcamach2! 

it is missing the part to check if it is a `list of int` or just a `int`. You can look at this example: https://github.com/dipy/dipy/blob/976217e547c4da6bfb210642adbb071c2ee40dd9/dipy/workflows/denoise.py#L310.

Also, it would be great if you could update the test : https://github.com/dipy/dipy/blob/976217e547c4da6bfb210642adbb071c2ee40dd9/dipy/workflows/tests/test_denoise.py#L43

Thank you!

**Comment by pcamach2** [comment]
Thank you for your review, @skoudoro!

Please let me know if anything else needs adjusting for this PR

**Review by skoudoro — APPROVED** [review]
Looks good to me! 

Waiting for the CI's to finish, then I will go ahead and merge it.

**Comment by skoudoro** [comment]
ok, all green, merging. Thanks again @pcamach2 !

**Pull request merged**